### DISCUSSION TITLE
PWM function. Help.

### DISCUSSION THREAD

**Discussion by Midnight83** [RavenSystem/esp-homekit-devices#1686] [MEPLHAA Scripts]
maybe anybody knows how can configuration pwm for fan accessories, I use lightbulb accessories as control to pc fan, ventilation fun, heater fan(wemos d1 mini) because they have pwm control function, but no visual aesthetic, maybe anybody share example?

Thanks.

**Comment by wahidulislamriyad** [comment]
PWM is only available for lightbulb. But it’s doesn’t even exist for Fan. RavenSystem should update the firmware to support PWM for Fans as like lightbulbs.

**Comment by RavenSystem** [comment]
You can use a Fan Service to control a non-HomeKit (with `"h":0`) Lightbulb Service using HAA Service Notifications and Wildcard Actions.
